internal/services/containers/container_group_resource.go

```diff
@@ -597,6 +598,11 @@ func resourceContainerGroupCreate(d *pluginsdk.ResourceData, meta interface{}) e
 	// https://docs.microsoft.com/en-us/azure/container-instances/container-instances-vnet#virtual-network-deployment-limitations
 	// https://docs.microsoft.com/en-us/azure/container-instances/container-instances-vnet#preview-limitations
 	if networkProfileID := d.Get("network_profile_id").(string); networkProfileID != "" {
+		// Avoid parallel provisioning if "network_profile_id" is given.
+		// See: https://github.com/hashicorp/terraform-provider-azurerm/issues/15025
+		locks.ByID(networkProfileID)
+		defer locks.UnlockByID(networkProfileID)
```

Comment:
since we're locking on the ID here we should probably update the validation to ensure the casing is consistent? e.g. add a validation func to `network_profile_id` in the schema?

presumably we'd also need this during the Update and Delete incase these are provisioned in parallel?